### PR TITLE
Update sf calc

### PR DESCRIPTION
**Pull request by amsnyder**
@ted80810 - what do you think about calculating the saltfront like this? It is getting all the values under 0.53, turning it into a dataframe, then getting the max salt value for each hour once it is already in a dataframe. That way there will only be one measurement per hour.

Really I would have liked to do the max piece in xarray, but I couldn't find enough documentation/examples to figure it out. This still takes about 5 mins for me to run, so while it's a little indirect, it gets the job done in the same amount of time.

### CODE REVIEW

**Review by ted80810 — COMMENTED** [review]
Gotcha. I think this looks great. The max function might work the same way as your code in line 78. I believe xarray wraps pandas' function "resample" as well, but like you said there aren't a lot of documented examples of this.

File: 01_fetch/src/fetch_coawst_model.py [hunk 1]
```diff
@@ -62,23 +62,21 @@ def salt_front_timeseries(ds, river_mile_coords_filepath, run_number):
...
     salt = salt.assign_coords({'dist_mile': dist_mile})
 
-    # locate saltfront
-    saltfront = salt.where(salt.salt < 0.53).where(salt.salt > 0.5)
-    saltfront_location = saltfront.where(saltfront.max('ocean_time'))
+    # locate values where salinity is at or below saltfront value (0.53)
+    saltfront = salt.where(salt.salt < 0.53)
 
     # subset salt variable
     saltfront_location = saltfront_location.salt
```
Comment (amsnyder) [inline 01_fetch/src/fetch_coawst_model.py:69]:
```suggestion
    saltfront_location = saltfront.salt
```
```suggestion
    saltfront_location = saltfront_location.salt
```

Comment (amsnyder) [inline 01_fetch/src/fetch_coawst_model.py:69]:
```suggestion
    saltfront_location = saltfront.salt
```
```suggestion
    saltfront_location = saltfront_location.salt
```

File: 01_fetch/src/fetch_coawst_model.py [hunk 2]
```diff
@@ -62,23 +62,21 @@ def salt_front_timeseries(ds, river_mile_coords_filepath, run_number):
...
-    df = df[df['salt'].notna()]
-    
     # drop points index column so we only have one index (ocean_time)
     df = df.droplevel(level=1)
 
+    # get actual maximum value to use as salt front location
+    df = df.resample('1H').max()
+
     # take daily average by averaging hourly location throughout day 
     df = df.resample('1D').mean()
```
Comment (amsnyder) [inline 01_fetch/src/fetch_coawst_model.py:82]:
```suggestion
    df = df.resample('1H').max()

    # take daily average by averaging hourly location throughout day 
    df = df.resample('1D').mean()
    
    # rename datetime column
    df.reset_index(inplace=True)
    df.rename({'ocean_time':'datetime'}, axis=1)

```

Comment (amsnyder) [inline 01_fetch/src/fetch_coawst_model.py:82]:
```suggestion
    df = df.resample('1H').max()

    # take daily average by averaging hourly location throughout day 
    df = df.resample('1D').mean()
    
    # rename datetime column
    df.reset_index(inplace=True)
    df.rename({'ocean_time':'datetime'}, axis=1)

```

File: 01_fetch/src/fetch_coawst_model.py [hunk 3]
```diff
@@ -62,25 +62,27 @@ def salt_front_timeseries(ds, river_mile_coords_filepath, run_number):
...
+    
+    # rename datetime column
+    df.reset_index(inplace=True)
+    df.rename({'ocean_time':'datetime'}, axis=1)
```
Comment (amsnyder) [inline 01_fetch/src/fetch_coawst_model.py:85]:
```suggestion
    s = df.groupby(pd.Grouper(freq='H'))['salt'].transform('max')
    df = df[df['salt'] == s]

    # take daily average by averaging hourly location throughout day 
    df = df.resample('1D').mean()

    # rename datetime column
    df.reset_index(inplace=True)
    df.rename({'ocean_time':'datetime'}, axis=1)
```

**Pull request merged**